service/src/main/kotlin/fi/espoo/evaka/daycare/PreschoolTerms.kt

```diff
@@ -0,0 +1,32 @@
+package fi.espoo.evaka.daycare
+
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.domain.ClosedPeriod
+import fi.espoo.evaka.shared.utils.zoneId
+import org.jdbi.v3.core.kotlin.mapTo
+import java.time.LocalDate
+
+data class PreschoolTerm(
+    val finnishPreschool: ClosedPeriod,
+    val swedishPreschool: ClosedPeriod,
+    val extendedTerm: ClosedPeriod,
```

Comment:
So what exactly is the purpose of each of these? I can't see any places that use the `finnishPreschool` or `swedishPreschool` values, or explanation where they would/should be used.

Reply:
I'll need to add these comments to code as well, but from domain perspective their meaning is:
- finnishPreschool: The period during which suva arranges preschool.
- swedishPreschool: The period during which svebi arranges preschool
- extendedTerm: Children going to preschool may start in connected daycare already before the actual preschool starts, essentially being normal daycare with discounted price. This period defines when the connected daycare is available.
- applicationPeriod: The official application period. The end date is not enforced though, but applications are accepted until end of term.

Not all of this data is used yet, but 
- In the new react frontend we likely want to replace some hard-coded dates with these e.g. in the application instruction texts. 
- At least in placement dialog there's hard-coded term-based logic that should use these (next PR)
- Later we likely want a validation logic that if user applies to preschool _only_, then preferredStartDate must be within finnish/swedishPreschool, while if they also apply for connected daycare, then preferredStartDate must be within extendedTerm.

So yes, adding all these is somewhat premature, but I would argue they are needed soon enough.

Reply:
OK :+1: Is it reasonable to assume suva & svebi are the only special cases even for other municipalities, i.e. that we shouldn't separate them into their own "model" referencing the main model with _just_ the `extendedTerm`? This does feel relatively simple to split later on if necessary but just wondering.

Reply:
Yeah those are awkwardly Espoo specific and may need some refactoring later.. But I'm not yet sure what exactly so maybe best to figure out later.